lxd/images.go

```diff
@@ -1729,8 +1775,14 @@ func imagesGet(d *Daemon, r *http.Request) response.Response {
 
 	request.SetCtxValue(r, request.CtxEffectiveProjectName, effectiveProjectName)
 
-	// If the caller is not trusted, we only want to list public images.
-	publicOnly := !auth.IsTrusted(r.Context())
+	// If the caller is not trusted, we only want to list public images in the default project.
+	trusted := auth.IsTrusted(r.Context())
+	publicOnly := !trusted
+
+	// Untrusted callers can't request images from all projects or projects other than default.
+	if !trusted && (allProjects || projectName != api.ProjectDefaultName) {
+		return response.Forbidden(errors.New("Untrusted requests can only access public images in the default project"))
```

Comment:
I'm still not sure on this error message (sorry I know I suggested it!). Suggest changing
- `can` to `may`
- `requests` to `callers` (or `users` perhaps... I'm not sure).

Reply:
I think `callers` is more consistent with the language used in comments throughout `lxd/images.go`.